src/tinypeel/tinypeel.py

```diff
@@ -213,19 +213,160 @@ def generateSingleLocusSegregation(peelingInfo, pedigree, args):
         peelingInfo.segregation[:, :, :] = 0.25
 
 
+def get_probability_options():
+    parse_dictionary = dict()
+    parse_dictionary["geno_error_prob"] = lambda parser: parser.add_argument(
+        "-geno_error_prob",
+        default=0.0001,
+        required=False,
+        type=float,
+        help="Genotyping error rate. Default: 0.0001.",
+    )
+    parse_dictionary["seq_error_prob"] = lambda parser: parser.add_argument(
+        "-seq_error_prob",
+        default=0.001,
+        required=False,
+        type=float,
+        help="Sequencing error rate. Default: 0.001.",
+    )
+
+    return parse_dictionary
+
+
+def get_input_options():
+    parse_dictionary = dict()
+    parse_dictionary["bfile"] = lambda parser: parser.add_argument(
+        "-plink_file",
+        default=None,
+        required=False,
+        type=str,
+        nargs="*",
+        help="plink (binary) file(s).",
+    )
+    parse_dictionary["genotypes"] = lambda parser: parser.add_argument(
+        "-geno_file",
+        default=None,
+        required=False,
+        type=str,
+        nargs="*",
+        help="Genotype File(s) in AlphaGenes format.",
+    )
+    parse_dictionary["reference"] = lambda parser: parser.add_argument(
+        "-reference",
+        default=None,
+        required=False,
+        type=str,
+        nargs="*",
+        help="A haplotype reference panel in AlphaGenes format.",
+    )
+    parse_dictionary["seqfile"] = lambda parser: parser.add_argument(
+        "-seq_file",
+        default=None,
+        required=False,
+        type=str,
+        nargs="*",
+        help="Sequence allele read count file(s).",
+    )
+    parse_dictionary["pedigree"] = lambda parser: parser.add_argument(
+        "-ped_file",
+        default=None,
+        required=False,
+        type=str,
+        nargs="*",
+        help="Pedigree file(s) in AlphaGenes format.",
+    )
+    parse_dictionary["phasefile"] = lambda parser: parser.add_argument(
+        "-hap_file",
+        default=None,
+        required=False,
+        type=str,
+        nargs="*",
+        help="A haplotype file in AlphaGenes format.",
+    )
+    parse_dictionary["startsnp"] = lambda parser: parser.add_argument(
+        "-start_snp",
+        default=None,
+        required=False,
+        type=int,
+        help="The first marker to consider. The first marker in the file is marker '1'. Default: 1.",
+    )
+    parse_dictionary["stopsnp"] = lambda parser: parser.add_argument(
+        "-stop_snp",
+        default=None,
+        required=False,
+        type=int,
+        help="The last marker to consider. Default: all markers considered.",
+    )
+    parse_dictionary["seed"] = lambda parser: parser.add_argument(
+        "-seed",
+        default=None,
+        required=False,
+        type=int,
+        help="A random seed to use for debugging.",
+    )
+
+    return parse_dictionary
```

Comment:
@AprilYUZhang 
Even though you keep the keys of the `parse_dictionary` unchanged, you still need to change how you access the values via `args` inside the main program. 

I did a test program, it basically does the same thing as those functions in `tinyhouse.InputOutput`

```python
import argparse
import sys

parser = argparse.ArgumentParser(description = 'Test')

def add_arguments_from_dictionary(parser, arg_dict, options = None):
    if options is None:
        for key, value in arg_dict.items():
            value(parser)
    else:
        for option in options:
            if option in arg_dict:
                arg_dict[option](parser)
            else:
                print("Option not found:", option, arg_dict)

def get_input_options():

    parse_dictionary = dict()
    parse_dictionary["bfile"] = lambda parser: parser.add_argument(
        '-plink_file',   
        default=None, 
        required=False, 
        type=str, 
        nargs="*", 
        help='A file in plink (binary) format. Only stable on Linux).'
        )
    
    return parse_dictionary

input_parser = parser.add_argument_group("Input Options")
add_arguments_from_dictionary(
        input_parser,
        get_input_options(),
        options=[
            "bfile"
        ],
)

args = sys.argv[1:]
args = parser.parse_args(args)
print(f"{args.plink_file=}")
print(f"{args.bfile=}")
```
And I get 
```bash
(base) evie@Xingers-MacBook-Air test % python /Users/evie/test/src/test/testing.py -plink_file "FILE NAME"
args.plink_file=['FILE NAME']
Traceback (most recent call last):
  File "/Users/evie/test/src/test/testing.py", line 43, in <module>
    print(f"{args.bfile=}")
AttributeError: 'Namespace' object has no attribute 'bfile'
```
So the `args` can only know `plink_file` as the name in the command line `plink_file` but not the dictionary key `bfile`.

You still need to change the places where the option names are changed in the main program. I think one way to do this is by redefining the attributes of the `args` 
```python
args = sys.argv[1:]
args = parser.parse_args(args)
args.bfile = args.plink_file
print(f"{args.bfile=}")
```
And with this change I can get
```bash
(base) evie@Xingers-MacBook-Air test % python /Users/evie/test/src/test/testing.py -plink_file "FILE NAME"
args.bfile=['FILE NAME']
```

Reply:
Maybe today we can talk about it, but why you need use args.bfile in the test? I'm a little confused. Because I don't know what is wrong in the main programs.

Reply:
@AprilYUZhang We don't need `args.bfile` in the test for now, I just use it as an example to explain why I think you need to add a few more lines to make the code work.